docs/src/test-api/class-testoptions.md

```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
 
 Options used to create the context, as passed to [`method: Browser.newContext`]. Specific options like [`property: TestOptions.viewport`] take priority over this.
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    contextOptions: {
+      viewport: viewport: { width: 1280, height: 720 },
+    },
+  },
+});
+```
+
 ## property: TestOptions.deviceScaleFactor = %%-context-option-devicescalefactor-%%
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    deviceScaleFactor: 2
```

Comment:
This should also be in a project. We can also say "use device" instead and link to devices doc.